core/org.wso2.carbon.user.core/src/test/java/org/wso2/carbon/user/core/util/DatabaseUtilTest.java

```diff
@@ -0,0 +1,260 @@
+        log.info("Unhandled SQLRecoverableException");
+        exceptionNotHandled = true;
+        } catch (SQLException e) {
+        // this shouldn't happen
+        Assert.fail();
+        } finally {
+            Assert.assertFalse(exceptionNotHandled);
+        }
+    }
+
+    @Test
+    public void createUserStoreDataSource() throws Exception {
+
+        when(realmConfig.isLogAbandoned()).thenReturn(true);
+        when(realmConfig.isRemoveAbandoned()).thenReturn(true);
+        when(realmConfig.getRemoveAbandonedTimeout()).thenReturn(100);
+        DatabaseUtil.createUserStoreDataSource(realmConfig);
+
+        Assert.assertTrue(realmConfig.isLogAbandoned());
+        Assert.assertTrue(realmConfig.isRemoveAbandoned());
+        Assert.assertEquals(100, realmConfig.getRemoveAbandonedTimeout());
+        Assert.assertNotNull(datasource.getConnection());
+    }
+
+    @Test
+    public void nullCreateUserStoreDataSource() throws Exception {
+
+        DatabaseUtil.createUserStoreDataSource(realmConfig);
+
+        Assert.assertFalse(realmConfig.isLogAbandoned());
+        Assert.assertFalse(realmConfig.isRemoveAbandoned());
+        Assert.assertEquals(0, realmConfig.getRemoveAbandonedTimeout());
+
+        Assert.assertEquals(realmConfig.isLogAbandoned(), poolProperties.isLogAbandoned());
+        Assert.assertEquals(realmConfig.isRemoveAbandoned(), poolProperties.isRemoveAbandoned());
+        Assert.assertEquals(realmConfig.getRemoveAbandonedTimeout(), poolProperties.getRemoveAbandonedTimeout());
+    }
+
+    @Test
+    public void getStringValuesFromDatabaseShouldReturnValues() throws Exception{
+
+        ResultSetMocker resultSetMocker = new ResultSetMocker().invoke();
+        String sqlStmt = resultSetMocker.getSqlStmt();
+        List<Object> params = resultSetMocker.getParams();
+        List<Object> resultSetAnswers = resultSetMocker.getResultSetAnswers();
+
+        String[] results =   DatabaseUtil.getStringValuesFromDatabase(conn, sqlStmt, params.get(0), params.get(1), params.get(2));
+
+        Assert.assertEquals( resultSetAnswers.get(0), results[0]);
+        Assert.assertEquals(resultSetAnswers.get(1), results[1]);
+        Assert.assertEquals(resultSetAnswers.get(2), results[2]);
+    }
+
+
+    @Test (expected = UserStoreException.class)
+    public void getStringValuesFromDatabaseShouldHandleSQLRecoverableException() throws Exception{
+
+        // cannot handle sqlerecoverable for resultsets
+        ResultSetMocker resultSetMocker = new ResultSetMocker().invoke();
+        String sqlStmt = resultSetMocker.getSqlStmt();
+        List<Object> params = resultSetMocker.getParams();
+        Mockito.doThrow(new SQLRecoverableException("throw SQLRecoverableException")).when(resultSetMocker.get()).next();
+
+        String[] results =   DatabaseUtil.getStringValuesFromDatabase(conn, sqlStmt, params.get(0), params.get(1), params.get(2));
+        Assert.fail("Can not handle SQLRecoverableError" );

```

Comment:
It is not a typo.  Although, both “can not” and “cannot” are observed in writing, “cannot” is the correct (and more acceptable) spelling.
http://www.dailywritingtips.com/cannot-or-can-not/

Spelling in a comment, in a unit test, seems like a trivial reason to hold up a pull request.  Is there some other reason?  Else, would you otherwise proceed?

Thanks,
Lorren


From: Imesh Guaratne [mailto:notifications@github.com]
Sent: Sunday, July 24, 2016 8:23 PM
To: wso2/carbon-kernel <carbon-kernel@noreply.github.com>
Cc: Kay, Lorren (Plymouth Meeting 2) <LKay@us.imshealth.com>; Mention <mention@noreply.github.com>
Subject: Re: [wso2/carbon-kernel] handle SqlrecoverableException (#1068)


In core/org.wso2.carbon.user.core/src/test/java/org/wso2/carbon/user/core/util/DatabaseUtilTest.java<https://github.com/wso2/carbon-kernel/pull/1068#discussion_r71998987>:

> +        Assert.assertEquals(resultSetAnswers.get(1), results[1]);

> +        Assert.assertEquals(resultSetAnswers.get(2), results[2]);

> +    }

> +

> +

> +    @Test (expected = UserStoreException.class)

> +    public void getStringValuesFromDatabaseShouldHandleSQLRecoverableException() throws Exception{

> +

> +        // cannot handle sqlerecoverable for resultsets

> +        ResultSetMocker resultSetMocker = new ResultSetMocker().invoke();

> +        String sqlStmt = resultSetMocker.getSqlStmt();

> +        List<Object> params = resultSetMocker.getParams();

> +        Mockito.doThrow(new SQLRecoverableException("throw SQLRecoverableException")).when(resultSetMocker.get()).next();

> +

> +        String[] results =   DatabaseUtil.getStringValuesFromDatabase(conn, sqlStmt, params.get(0), params.get(1), params.get(2));

> +        Assert.fail("Can not handle SQLRecoverableError" );

Typo: Can not => Cannot

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/wso2/carbon-kernel/pull/1068/files/c3edc4a17882cec9b9b5e5ee3c319f825c4a6b42#r71998987>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AD9dJIYby4XZSNYBjDZgK-neQVWOU3MHks5qZAHpgaJpZM4I-GcD>.